src/main/java/org/reactome/addlinks/fileprocessors/MGIFileProcessor.java

```diff
@@ -0,0 +1,57 @@
+package org.reactome.addlinks.fileprocessors;
+
+import org.reactome.addlinks.EnsemblBiomartUtil;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class MGIFileProcessor extends FileProcessor{
+
+    public MGIFileProcessor(String processorName)
+    {
+        super(processorName);
+    }
+
+    public MGIFileProcessor()
+    {
+        super();
+    }
+
+    private static final int mgiIdentifierIndex = 0;
+    private static final int uniprotIdentifiersIndex = 6;
+
+    /**
+     * Build map of UniProt identifiers to Mouse Genome Informatics (MGI) identifiers that is used to create MGI cross-references in database.
+     * @return - Map<String, List<String>
+     */
+    @Override
+    public Map<String, List<String>> getIdMappingsFromFile()
+    {
+        Map<String, List<String>> mappings = new HashMap<>();
+        Path inputFilePath = Paths.get(this.pathToFile.toAbsolutePath().toString().replace(".gz", ""));
+        List<String> lines = new ArrayList<>();
+        try {
+            lines = EnsemblBiomartUtil.getLinesFromFile(inputFilePath, false);
+        } catch (IOException e) {
+            logger.error("Error reading file ({}): {}", inputFilePath.toString(), e.getMessage());
+            e.printStackTrace();
+        }
+
+        for (String line : lines) {
+            List<String> tabSplit = Arrays.asList(line.split("\t"));
+            if (EnsemblBiomartUtil.necessaryColumnPresent(tabSplit, uniprotIdentifiersIndex)) {
+
+                String mgiId = tabSplit.get(mgiIdentifierIndex).split(":")[1];
```

Comment:
For context of taking the second element of the array, maybe this logic could be extracted to a helper method:

```
String mgiId = getMgiIDWithoutPrefix(tabSplit.get(mgiIdentifierIndex));

private String getMgiIDWithoutPrefix(String fullMgiId) {
    return fullMgiId.split(":")[1];
}
```

Reply:
Done